deepchem/feat/smiles_tokenizer.py

```diff
@@ -0,0 +1,292 @@
+# Requriments - transformers, tokenizers
+# Right now, the Smiles Tokenizer uses an exiesting vocab file from rxnfp that is fairly comprehensive and from the USPTO dataset.
+# The vocab may be expanded in the near future
+
+import collections
+import logging
+import os
+import re
+import numpy as np
+import pkg_resources
+from typing import List
+from transformers import BertTokenizer
+
+# export
+SMI_REGEX_PATTERN = r"""(\[[^\]]+]|Br?|Cl?|N|O|S|P|F|I|b|c|n|o|s|p|\(|\)|\.|=|
+#|-|\+|\\|\/|:|~|@|\?|>>?|\*|\$|\%[0-9]{2}|[0-9])"""
+
+def get_default_tokenizer():
+    default_vocab_path = (
+        pkg_resources.resource_filename(
+            "chemberta",
+            "tokenizers/vocab.txt"
+        )
+    )
+    return SmilesTokenizer(default_vocab_path)
+
+class SmilesTokenizer(BertTokenizer):
+    """
+    Creates the SmilesTokenizer class. The tokenizer heavily inherits from the BERT
+    WordPieceTokenizer implementation found in Huggingface's transformers library.
+
+    Please see https://github.com/huggingface/transformers
+    and https://github.com/rxn4chemistry/rxnfp for more details.
+
+    Examples
+    --------
+
+    >>> from deepchem.feat.smiles_tokenizer import SmilesTokenizer
+    >>> from transformers import RobertaForMaskedLM
+
+    >>> current_dir = os.path.dirname(os.path.realpath(__file__))
+    >>> vocab_path = os.path.join(current_dir, 'data', 'vocab.txt')
+
+    >>> tokenizer = SmilesTokenizer(vocab_path)
+    >>> print(tokenizer.encode("CCC(CC)COC(=O)[C@H](C)N[P@](=O)(OC[C@H]1O[C@](C#N)([C@H](O)[C@@H]1O)C1=CC=C2N1N=CN=C2N)OC1=CC=CC=C1"))
+
+
+    References
+    ----------
+    .. [1]  Schwaller, Philippe; Probst, Daniel; Vaucher, Alain C.; Nair, Vishnu H; Kreutter, David;
+            Laino, Teodoro; et al. (2019): Mapping the Space of Chemical Reactions using Attention-Based Neural
+            Networks. ChemRxiv. Preprint. https://doi.org/10.26434/chemrxiv.9897365.v3
+    Notes
+    ----
+    This class requires huggingface's transformers and tokenizers libraries to be installed.
+
+    """
+    def __init__(
+            self,
+            vocab_file: str='',
+            # unk_token="[UNK]",
```

Comment:
Should we still have these commented out arguments here?

Reply:
might be good to keep to preserve how the original wordpiece tokenizer works!

Reply:
can remove if needed

Reply:
Ah, so it's documentation about the vocab file format! Makes sense. Can you move this into into the docstring below so the vocab file format shows up in the rendered docs?